### DISCUSSION TITLE
How to pass customized arguments to the fit() function of the learner

### DISCUSSION THREAD

**Discussion by chungdz** [DoubleML/doubleml-for-py#207] [General] [answered]
Hi guys, I want to apply catboost as the regressor learner as there're categorical features in the x_cols to fit. I need to identify which columns are categorical data. And the argument to identify them, cat_features, is in the fit() function of the catboost. I just want to know how to pass this argument to the fit() function of the learner? I try to find positions where DoubleMLPLR calls fit() function of the learner but I think they're hidden in the code file deeply such that I still didn't find the positions.

**Comment by SvenKlaassen** [comment, marked as answer]
Thank you for your interest in the package.
Just to clarify. Isn't it possible use the `cat_features` argument in `CatBoostRegressor` ([documentation](https://catboost.ai/en/docs/concepts/python-reference_catboostregressor#cat_features))?

This should be possible to use in `DoubleMLPLR`

```
from catboost import CatBoostRegressor
# Initialize data
cat_features = [0, 1]
train_data = [["a", "b", 1, 4, 5, 6],
              ["a", "b", 4, 5, 6, 7],
              ["c", "d", 30, 40, 50, 60]]
train_labels = [1.2, 1.1, -1.3]
eval_data = [["a", "b", 2, 4, 6, 8],
             ["a", "d", 1, 4, 50, 60]]

# Initialize CatBoostRegressor
model = CatBoostRegressor(iterations=2,
                          learning_rate=1,
                          depth=2,
                          cat_features=cat_features)
# Fit model
model.fit(train_data, train_labels)
# Get predicted values
preds = model.predict(eval_data)
```

If this is not what you are looking for a minimal reproducable example would be helpful.

You can also create your own learners (add additional arguments etc.) as long as they fulfill the [minimum requirements](https://docs.doubleml.org/dev/guide/learners.html#minimum-requirements-for-learners).

In case you still want to check where `fit()` is used:
The predictions of the `DoubleMLPLR` model are generated here
https://github.com/DoubleML/doubleml-for-py/blob/df617fdc2678e9882006a95620bf1246ac2ffa30/doubleml/double_ml_plr.py#L176
which mostly uses  `cross_val_predict` from `scikit-learn`
https://github.com/DoubleML/doubleml-for-py/blob/df617fdc2678e9882006a95620bf1246ac2ffa30/doubleml/_utils.py#L63
